aiutils/tftools/layers.py

```diff
@@ -81,3 +81,41 @@ def conv2d(input,
                                    strides=strides,
                                    padding=padding) + b)
     return output
+
+
+def batch_norm(input, name):
+    """ Batch norm layer helper.
+    
+    Gets mean and variance, and creates offset and scale parameters with good initial values.
+    Then applies the batch_normalization op.
+    
+    Args:
+      input (tensor): Input to the layer. 
+        Should have shape `[batch, in_dim]` or `[batch, in_height, in_width, in_dim]`.
+        Must be one of the following types: `float32`, `float64`.
+      name (str): Name used by the `tf.variable_scope`.
+      
+    Returns:
+      output (tensor): Batch normalized activations.
+        Will have the same shape as input.
+    """
+    rank = len(input.get_shape().as_list())
+    in_dim = input.get_shape().as_list()[-1]
+    if rank == 2:
+        axes = [0]
+    elif rank == 4:
+        axes = [0, 1, 2]
+    else:
+        raise ValueError('rank must be 2 or 4.')
```

Comment:
I think an assertion before the conditional is equivalent to the current code, since all assertions do is to raise a different type of exception in the end. It would add a redundant check for 2 or 4 dimensions instead. 

On the other hand, if you want to throw it into the logs, just add an `import logging` at the top of the file and maybe add a log entry here before 109 with something like `logging.error("rank was not 2 or 4")`